src/main/webapp/js/datahandlers/collegepagedatahandler.js

```diff
@@ -87,6 +89,23 @@ class CollegeDataHandler extends SinglePageDataHandler {
       cumulativeMedianDebt: element[MEDIAN_DEBT].toString()
     };
   }
+
+  async findScholarships(id) {
+    try {
+      let scholarshipResponse = await fetch(`${FIND_SCHOLARSHIP_ENDPOINT}?id=${id}`);
+      let scholarshipJson = await scholarshipResponse.json();
+      return scholarshipJson.map((e) => this.formatScholarshipListButton(e));
+    } catch(e) {
+      throw new Error(`Cannot get scholarship json ${e}`);
+    }
+  }
+
+  formatScholarshipListButton(elem) {
+    return {
+      'name' : elem[SCHOLARSHIP_NAME],
+      'id' : elem['id'],
```

Comment:
The rendered template shows undefined if I change the strings here to constant variables. I'm not sure how to fix it.

Reply:
please debug and find out.